### PR TITLE
New permission center button

### PR DESCRIPTION
**Pull request by tomasstrba**
Task/Issue URL: https://app.asana.com/1/137249556945/project/1148564399326804/task/1212014856091174?focus=true
CC: 

### Description
Adding new feature flag, and new permission center button

### Testing Steps

**Test disabled feature flag**
1. Make sure `newPermissionView` feature flag is disabled
2. Visit https://permission.site/ and smoke test following permissions: camera, location, and microphone
3. Test popups (Execute open.window("https://wikipedia.org" in JS Console))
4. Test external schemes (https://www.w3schools.com/tags/tryit.asp?filename=tryhtml_link_mailto)
5. Verify granting and managing permissions work as expected

**Test new notification center button**
1. Enable `newPermissionView` feature flag
2. Visit https://permission.site/ and test following permissions: camera, location, and microphone
3. Verify the legacy authorization flows are attached to the new permission center button
4. Verify the permission center button changes icon when there is a pending request, otherwise there is the default button
5. Test popups (Execute open.window("https://wikipedia.org" in JS Console))
6. Test external schemes (https://www.w3schools.com/tags/tryit.asp?filename=tryhtml_link_mailto)

_Note: Adjusting authorization flows and implementing the center itself will be delivered in follow-up PRs_

### Notes to Reviewer

---
###### Internal references:
[Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f) | [Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552) | [Tech Design Template](https://app.asana.com/0/59792373528535/184709971311943)

---

> [!NOTE]
> Introduces a feature-flagged Permission Center button in the address bar, routes permission popovers to it, and adds new permissions icons and supporting helpers.
> 
> - **macOS UI**:
>   - Adds `permissionCenterButton` to the address bar (`AddressBarButtonsViewController`, `NavigationBar.storyboard`), including size/visibility logic and icon swap to bell when a request is pending.
>   - Centralizes permission handling: new path uses `permissionCenterButton` for popovers (including popups/external scheme tweaks), with fallback to legacy per-permission buttons when the flag is off.
>   - Updates subscriptions and button update flows (`updateAllPermissionButtons`, popover show/close handling), plus Tab-level helper `TabViewModel.shouldShowPermissionCenterButton`.
> - **Feature Flags**:
>   - Adds `MacOSBrowserConfigSubfeature.newPermissionView` and `FeatureFlag.newPermissionView`; maps to remote releasable source and enables local overriding.
> - **Design Resources**:
>   - Adds permissions glyphs and assets: `permissions16`, `permissionsLocation16`, `permissionsLocationSolid16`, `permissionsNotifications16`, `permissionsNotificationsSolid16`; exposes via `DesignSystemImages.Glyphs.Size16`.
> - **Permissions Models**:
>   - Adds `PermissionState.isReloading` and `PermissionType.isPopups` helpers to support new logic.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 06fc50b71b33459a7069dbd4315a5875ffc94352. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: macOS/DuckDuckGo/NavigationBar/View/AddressBarButtonsViewController.swift [hunk 1]
```diff
@@ -1511,6 +1594,10 @@ final class AddressBarButtonsViewController: NSViewController {
         }
     }
 
+    @IBAction func permissionCenterButtonAction(_ sender: Any) {
+
```
[the remark this answers is not in the document]
Comment (tomasstrba) [reply macOS/DuckDuckGo/NavigationBar/View/AddressBarButtonsViewController.swift:1607]: lol

File: macOS/DuckDuckGo/NavigationBar/View/AddressBarButtonsViewController.swift [hunk 2]
```diff
@@ -1397,36 +1479,40 @@ final class AddressBarButtonsViewController: NSViewController {
...
+                    popover = popupBlockedPopoverCreatingIfNeeded()
+                case .externalScheme:
+                    button = externalSchemeButton
+                    query.shouldShowAlwaysAllowCheckbox = true
+                    query.shouldShowCancelInsteadOfDeny = true
+                default:
+                    assertionFailure("Unexpected permissions")
+                    query.handleDecision(grant: false)
+                    return
+                }
```
[the remark this answers is not in the document]
Comment (tomasstrba) [reply macOS/DuckDuckGo/NavigationBar/View/AddressBarButtonsViewController.swift:1512]: good point 👍

File: macOS/DuckDuckGo/Assets.xcassets/Images/PermissionCenterIcon.imageset/Contents.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (jotaemepereira) [inline macOS/DuckDuckGo/Assets.xcassets/Images/PermissionCenterIcon.imageset/Contents.json]: ℹ️  @tomasstrba, we should use `DesignResourcesKitIcons` for these assets. 

Here is the repo: https://dub.duckduckgo.com/duckduckgo/Icons, which should have the icons if they are not already in the Assets.catalog in the DRKI package.

File: macOS/DuckDuckGo/NavigationBar/View/AddressBarButtonsViewController.swift [hunk 4]
```diff
@@ -650,6 +659,76 @@ final class AddressBarButtonsViewController: NSViewController {
...
+        guard let tabViewModel else { return }
+
+        guard featureFlagger.isFeatureOn(.newPermissionView) else {
+            permissionCenterButton.isShown = false
+            return
+        }
+
+        // Show permission buttons when there's a requested permission on NTP even if address bar is focused,
+        // since NTP has the address bar focused by default
+        let hasRequestedPermission = tabViewModel.usedPermissions.values.contains(where: { $0.isRequested
```
Comment (jotaemepereira) [inline macOS/DuckDuckGo/NavigationBar/View/AddressBarButtonsViewController.swift]: 💡  `hasRequestedPermission`, `shouldShowWhileFocused` and `isAnyPermissionPresent` values all come from the tab view model, I think we should keep it there where we can easily add unit tests for the different states.

For example, we could have the following in the view model:

```swift
    func shouldShowPermissionsButton(isTextFieldEditorFirstResponder: Bool,
                                     isAnyTrackerAnimationPlaying: Bool) -> Bool {
        // Show permission buttons when there's a requested permission on NTP even if address bar is focused,
        // since NTP has the address bar focused by default
        let hasRequestedPermission = usedPermissions.values.contains(where: { $0.isRequested
        })
        let shouldShowWhileFocused = (tab.content == .newtab) && hasRequestedPermission
        let isAnyPermissionPresent = usedPermissions.values.contains(where: {
            !$0.isReloading
        })

        return (shouldShowWhileFocused ||
                (!isTextFieldEditorFirstResponder && hasRequestedPermission) ||
                isAnyPermissionPresent)
        && !isAnyTrackerAnimationPlaying
        && !isShowingErrorPage
    }
```

**Comment by tomasstrba** [comment]
Thank you, @jotaemepereira.

1) The icons are not in the repository yet. I’m waiting for Fran to add them. In the meantime I used the globe and heart icons from the kit, and I’ll swap them once the correct ones are available. Hopefully this works better for you than adding them directly to Assets
2) The settings dashboard icon edge case is fixed.
3) I refactored the ViewModel based on your suggestion.

It's ready for another look

**Review by jotaemepereira — APPROVED** [review]
Looks great now 👍🏼 .

If you can add some unit tests to the `tabViewModel.shouldShowPermissionCenterButton` that would be great
